src/runtime_src/core/include/xcl_app_debug.h

```diff
@@ -43,14 +43,68 @@ extern "C" {
 #endif
 
 /************************ AIM Debug Counters ********************************/
-#define XAIM_DEBUG_SAMPLE_COUNTERS_PER_SLOT     9
+#define XAIM_DEBUG_SAMPLE_COUNTERS_PER_SLOT           9
+#define XAIM_TOTAL_DEBUG_SAMPLE_COUNTERS_PER_SLOT     13
+
+#define XAIM_WRITE_BYTES_INDEX         0
+#define XAIM_WRITE_TRANX_INDEX         1
+#define XAIM_READ_BYTES_INDEX          2
+#define XAIM_READ_TRANX_INDEX          3
+#define XAIM_OUTSTANDING_COUNT_INDEX   4
+#define XAIM_WRITE_LAST_ADDRESS_INDEX  5
+#define XAIM_WRITE_LAST_DATA_INDEX     6
+#define XAIM_READ_LAST_ADDRESS_INDEX   7
+#define XAIM_READ_LAST_DATA_INDEX      8
+
+#define XAIM_IOCTL_WRITE_BYTES_INDEX         0
+#define XAIM_IOCTL_WRITE_TRANX_INDEX         1
+#define XAIM_IOCTL_WRITE_LATENCY_INDEX       2
+#define XAIM_IOCTL_WRITE_BUSY_CYCLES_INDEX   3
+#define XAIM_IOCTL_READ_BYTES_INDEX          4
+#define XAIM_IOCTL_READ_TRANX_INDEX          5
+#define XAIM_IOCTL_READ_LATENCY_INDEX        6
+#define XAIM_IOCTL_READ_BUSY_CYCLES_INDEX    7
+#define XAIM_IOCTL_OUTSTANDING_COUNT_INDEX   8
+#define XAIM_IOCTL_WRITE_LAST_ADDRESS_INDEX  9
+#define XAIM_IOCTL_WRITE_LAST_DATA_INDEX     10
+#define XAIM_IOCTL_READ_LAST_ADDRESS_INDEX   11
+#define XAIM_IOCTL_READ_LAST_DATA_INDEX      12
 
 /************************ AM Debug Counters ********************************/
 #define XAM_DEBUG_SAMPLE_COUNTERS_PER_SLOT     8
+#define XAM_TOTAL_DEBUG_COUNTERS_PER_SLOT      10
+
+#define XAM_ACCEL_EXECUTION_COUNT_INDEX       0
+#define XAM_ACCEL_EXECUTION_CYCLES_INDEX      1
+#define XAM_ACCEL_STALL_INT_INDEX             2
+#define XAM_ACCEL_STALL_STR_INDEX             3
+#define XAM_ACCEL_STALL_EXT_INDEX             4
+#define XAM_ACCEL_MIN_EXECUTION_CYCLES_INDEX  5
+#define XAM_ACCEL_MAX_EXECUTION_CYCLES_INDEX  6
+#define XAM_ACCEL_TOTAL_CU_START_INDEX        7
+
+#define XAM_IOCTL_EXECUTION_COUNT_INDEX       0
+#define XAM_IOCTL_START_COUNT_INDEX           1
+#define XAM_IOCTL_EXECUTION_CYCLES_INDEX      2
+#define XAM_IOCTL_STALL_INT_INDEX             3
+#define XAM_IOCTL_STALL_STR_INDEX             4
+#define XAM_IOCTL_STALL_EXT_INDEX             5
+#define XAM_IOCTL_BUSY_CYCLES_INDEX           6
+#define XAM_IOCTL_MAX_PARALLEL_ITR_INDEX      7
+#define XAM_IOCTL_MAX_EXECUTION_CYCLES_INDEX  8
+#define XAM_IOCTL_MIN_EXECUTION_CYCLES_INDEX  9
 
 /************************ ASM Debug Counters ********************************/
 #define XASM_DEBUG_SAMPLE_COUNTERS_PER_SLOT    5
 
+#define XASM_NUM_TRANX_INDEX      0
+#define XASM_DATA_BYTES_INDEX     1
+#define XASM_BUSY_CYCLES_INDEX    2
+#define XASM_STALL_CYCLES_INDEX   3
+#define XASM_STARVE_CYCLES_INDEX  4
```

Comment:
Actually, I have thought of a better solution. We can introduce new struct type for each monitor result and use it in device query methods and the get_*_counter_result methods. Like 
typedef struct {
  unsigned long long int WriteBytes;
  unsigned long long int WriteTranx;
  unsigned long long int ReadBytes;
  unsigned long long int ReadTranx;
  unsigned long long int OutStandCnts;
  unsigned long long int LastWriteAddr;
  unsigned long long int LastWriteData;
  unsigned long long int LastReadAddr;
  unsigned long long int LastReadData;
} xclAIMCounterResult;
Then we can refer the counter values with the member fields of the struct, instead of # defined constants or enums. I have already discussed this with @jvillarre and we have planned it for a new PR. The current PR is already too big and mainly addresses OS specific handling for accessing counter values for different monitor types.

Reply:
Oh I see. Then you can "map" the `struct` on the driver data or even copy into it and just access each field.
This is clever!